src/ir/block_builder.h

```diff
@@ -49,9 +49,11 @@ class BlockBuilder {
   // Adds an operation to the block and returns the operation.
   const Operation& AddOperation(const Operator& op,
                                 NamedAttributeMap attributes,
-                                NamedValueMap inputs) {
-    block_->operations_.push_back(std::make_unique<Operation>(
-        block_.get(), op, std::move(attributes), std::move(inputs)));
+                                NamedValueMap inputs,
+                                std::unique_ptr<Module> impl_module = nullptr) {
```

Comment:
Should we have this be a default argument, or would it be better to make callers explicitly state that the implementation is null?

Reply:
I think having the default argument as `nullptr` is fine as it it will be most common usage.

Reply:
OK, sounds good.